data-dashboard/lib/components/ButtonWithIcon/ButtonWithIcon.tsx

```diff
@@ -0,0 +1,57 @@
+import styled from 'styled-components';
+import React from 'react';
+import { CarbonIconType } from '@carbon/icons-react';
+
+interface Props {
+  text: string;
+  icon: CarbonIconType;
+  style?: React.CSSProperties;
+}
+
+const ButtonComponent = styled.button`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+
+  gap: 10px;
+
+  padding-left: 16px;
+  padding-right: 10px;
+
+  background-color: transparent;
+  border: 2px solid #0f62fe;
+  cursor: pointer;
+  height: 48px;
+  width: fit-content;
+
+  color: #0f62fe;
+  font-size: 14px;
+
+  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+
+  transition: all 0.2s ease-in-out;
+
+  &:hover {
+    background-color: #0f62fe;
+    color: white;
+  }
+
+  &:active {
+    background-color: #002d9c;
+    border: 2px solid #002d9c;
+  }
+`;
+
+export default function ButtonWithIcon({
+  text,
+  icon: Icon,
+  style = undefined,
+}: Props): JSX.Element {
+  return (
+    <ButtonComponent style={style}>
+      {text}
+      <Icon aria-label="Add" size={24} />
```

Comment:
I believe it would be best to make the aria-label a parameter or not have it at all. Because there could be more uses for this component beyond adding stuff it should not be made to have a single purpose.

Reply:
I also suggest making a boolean property to determine wether the Icon should be on the left or right side of the text

Reply:
I'll make the arial label a parametere, my bad, should be inlcuded with the Icon

Reply:
The left or right icon is out of scope for this PR